packages/grafana-ui/src/components/Stack/Stack.tsx

```diff
@@ -2,17 +2,28 @@ import React from 'react';
 
 import { ThemeSpacingTokens } from '@grafana/data';
 
-import { Direction, Flex } from '../Flex/Flex';
+import { Flex } from '../Flex/Flex';
+
+import { HorizontalStack } from './HorizontalStack';
+import { VerticalStack } from './VerticalStack';
 
 interface StackProps {
-  direction?: Direction;
+  direction?: 'horizontal' | 'vertical';
   gap?: ThemeSpacingTokens;
 }
 
-export const Stack = ({ gap = 1, direction = 'column', children }: React.PropsWithChildren<StackProps>) => {
-  return (
-    <Flex gap={gap} direction={direction}>
-      {children}
-    </Flex>
-  );
-};
+export const Stack = React.forwardRef<HTMLDivElement, React.PropsWithChildren<StackProps>>(
+  ({ gap = 1, direction = 'vertical', children }, ref) => {
+    return (
+      <Flex ref={ref} gap={gap} direction={direction === 'vertical' ? 'column' : 'row'} wrap="wrap">
+        {React.Children.map(children, (child) => (
+          <div>{child}</div>
```

Comment:
Wondering why do we need to wrap children in divs here?

Reply:
The child alignment doesn't work as we want to unless we do this. Also, this prevents weird customization of child elements in use cases where Flex is a more appropriate component. Basically the Stack component is very locked down by design.